src/store.js

```diff
@@ -0,0 +1,49 @@
+import { clone, forEach } from 'ramda'
+
+export class Store {
+  pendingSets = [] // [{ key, value, resolve }]
+  pendingGets = [] // [{ key, resolve }]
+
+  initialize = initialStore => {
+    this.store = clone(initialStore)
```

Comment:
Why clone and not store the `initialStore`?

Reply:
Hmmmm. The resoning when I wrote this was probably something like: if the consumer mutates `initialStore` after initialization then that could fuck everything up, buuuut since we only do all the initialization intenally we probably needn't bother.